frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-topology/PipelineTaskNode.tsx

```diff
@@ -45,27 +45,29 @@ const PipelineTaskNode: React.FunctionComponent<PipelineTaskNodeProps> = ({
   const detailsLevel = useDetailsLevel();
   const isFinallyTask = element.getType() === NodeType.FINALLY_NODE;
   let resources;
-  if (data?.task.taskRef.kind === ClusterTaskModel.kind) {
+  if (data?.task?.taskRef?.kind === ClusterTaskModel.kind) {
     resources = {
       kind: referenceForModel(ClusterTaskModel),
       name: data?.task.taskRef.name,
       prop: 'task',
     };
-  } else {
+  } else if (data?.task?.taskRef) {
     resources = {
       kind: referenceForModel(TaskModel),
       name: data?.task.taskRef.name,
-      namespace: data.pipeline.metadata.namespace,
+      namespace: data?.pipeline.metadata?.namespace,
       prop: 'task',
     };
   }
   const [task] = useK8sWatchResource<TaskKind>(resources);
 
-  const stepList = data?.task?.status?.steps || task?.spec?.steps || [];
+  const computedTask = task && Object.keys(task).length ? task : data?.task;
+  const stepList =
+    computedTask?.status?.steps || computedTask?.spec?.steps || computedTask?.taskSpec?.steps || [];
 
   const pipelineRunStatus = data?.pipelineRun && pipelineRunFilterReducer(data?.pipelineRun);
   const isSkipped = data?.pipelineRun?.status?.skippedTasks?.some(
-    (t) => t.name === data?.task.name,
+    (t) => t.name === computedTask.name,
   );
```

Comment:
Can computedTask be undefined? Then I would recommend:

```
  const isSkipped = !!(computedTask && data?.pipelineRun?.status?.skippedTasks?.some(
    (t) => t.name === data?.task.name,
    (t) => t.name === computedTask.name,
  ));
```

Reply:
Its is the main resource for this task node, it will not be undefined but I can add the check.